src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -721,11 +721,74 @@ private void modifyGenerator(Generator generator, GeneratorModificationInfos mod
+        boolean isVoltageRegulationOn = modificationInfos.getVoltageRegulationOn() != null
+                ? modificationInfos.getVoltageRegulationOn().getValue()
+                : generator.isVoltageRegulatorOn();
+        generator.setVoltageRegulatorOn(isVoltageRegulationOn);
+
+        if (isVoltageRegulationOn) {
+            Terminal terminal = null;
+            if (modificationInfos.getRegulatingTerminalId() != null
+                    && modificationInfos.getRegulatingTerminalType() != null
+                    && modificationInfos.getRegulatingTerminalVlId() != null) {
+                terminal = getTerminalFromIdentifiable(generator.getNetwork(),
+                        modificationInfos.getRegulatingTerminalId().getValue(),
+                        modificationInfos.getRegulatingTerminalType().getValue(),
+                        modificationInfos.getRegulatingTerminalVlId().getValue());
+            }
+
+            generator.setRegulatingTerminal(terminal);
+
+            if (modificationInfos.getQPercent() != null) {
+                generator.newExtension(CoordinatedReactiveControlAdderImpl.class).withQPercent(modificationInfos.getQPercent().getValue())
+                        .add();
+            }
+        } else {
+            if (modificationInfos.getReactivePowerSetpoint() != null) {
+                applyElementaryModifications(generator::setTargetQ, generator::getTargetQ, modificationInfos.getReactivePowerSetpoint(), subReporter, "Reactive power set point");
+            }
+        }
+
+        if (modificationInfos.getTransientReactance() != null
+                && modificationInfos.getStepUpTransformerReactance() != null) {
+            generator.newExtension(GeneratorShortCircuitAdder.class)
+                    .withDirectTransX(modificationInfos.getTransientReactance().getValue())
+                    .withStepUpTransformerX(modificationInfos.getStepUpTransformerReactance().getValue())
+                    .add();
+        }
+
+        if (modificationInfos.getMarginalCost() != null) {
+            generator.newExtension(GeneratorStartupAdder.class)
+                    .withMarginalCost(modificationInfos.getMarginalCost().getValue()).add();
+        }
+        boolean participate = modificationInfos.getParticipate() != null ? modificationInfos.getParticipate().getValue()
+                : modificationInfos.getDroop() != null ? true : false;
+        if (participate) {
+            generator.newExtension(ActivePowerControlAdder.class)
+                    .withParticipate(participate).withDroop(modificationInfos.getDroop().getValue())
+                    .add();
+        } else {
+            generator.newExtension(ActivePowerControlAdder.class)
+                    .withParticipate(participate).add();
+        }
+
+        if (Boolean.TRUE.equals(modificationInfos.getReactiveCapabilityCurve().getValue() && modificationInfos.getReactiveCapabilityCurvePoints() != null)) {
```

Comment:
I really don't like this IF statement, because at first glance we read : 
`if ( ReactiveCapabilityCurve().getValue() == TRUE && CurvePoints() != null )`
but instead it is : 
`if ( ( ReactiveCapabilityCurve().getValue() && CurvePoints() != null ) == TRUE )`
The logic is not the same, and the way it is written is very confusing.